src/CloudNative.CloudEvents/Http/CloudEventHttpContent.cs

```diff
@@ -39,6 +36,8 @@ public CloudEventContent(CloudEvent cloudEvent, ContentMode contentMode, ICloudE
                 return;
             }
 
+            // TODO: Shouldn't we use the formatter in all cases? If I have a JSON formatter and
+            // If we specify that the the data is a byte array, I'd expect to end up with a base64-encoded representation...
```

Comment:
I agree with this line of thinking. Seems better to avoid doing magic under the hood.

Reply:
After my frustrations with Uri over the last day or two, I'm even more convinced of that :)

I suspect we want to abstract out two concepts:

- "I know how to convert data to/from bytes, based on a certain set of types"
- "I know how to encode/decode structured events"

It gets a bit tricky in the middle, where the structured event encoding may or may not need a data converter.